sycl/test/devicelib/assert.cpp

```diff
@@ -99,21 +100,95 @@
 // Note that the work-item that hits the assert first may vary, since the order
 // of execution is undefined. We catch only the first one (whatever id it is).
 
+#ifndef PARENT_PROCESS
 #include <CL/sycl.hpp>
+#endif
+
 #include <array>
 #include <assert.h>
 #include <stdlib.h>
+
+#ifdef PARENT_PROCESS
+#include <string.h>
+#endif
+
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
 
+const int EXIT_SKIP_TEST = 42;
+
+#ifdef PARENT_PROCESS
+int main(int argc, char *argv[]) {
+  assert(argc > 1);
+
+  char **ChildArgv = new char *;
+  ChildArgv[0] = argv[1];
+
+  int Child = fork();
+
+  if (Child < 0) {
+    perror("Fork failed");
+    return 1;
+  }
+
+  if (!Child) {
+    int ExecFailed = execve(argv[1], ChildArgv, environ);
+
+    if (ExecFailed) {
+      perror("Execve failed");
+      return 1;
+    }
+
+    assert(false && "Unreachanble reached");
+  }
+
+  int status = 0;
+  waitpid(Child, &status, 0);
+  if (WIFEXITED(status) && WEXITSTATUS(status) == EXIT_SKIP_TEST) {
+    return 0;
+  }
+  if (getenv("SHOULD_CRASH")) {
+    if (!WIFSIGNALED(status)) {
+      fprintf(stderr, "error: process did not terminate by a signal\n");
+      return 1;
+    }
+  } else {
+    if (WIFSIGNALED(status)) {
+      fprintf(stderr, "error: process should not terminate\n");
```

Comment:
Curious to have `fprintf()` in C++ code and all over the place...
If you really dislike `std::cerr <<` at least you can include `<cstdio>` and use `fputs` in this case instead of diving into varargs processing.